src/codegen.cpp

```diff
@@ -1924,22 +1902,21 @@ static std::set<int> assigned_in_try(jl_array_t *stmts, int s, long l, int *pend
 static void mark_volatile_vars(jl_array_t *stmts, std::vector<jl_varinfo_t> &slots)
 {
     size_t slength = jl_array_dim0(stmts);
-    for(int i=0; i < (int)slength; i++) {
-        jl_value_t *st = jl_array_ptr_ref(stmts,i);
+    for (int i = 0; i < (int)slength; i++) {
+        jl_value_t *st = jl_array_ptr_ref(stmts, i);
         if (jl_is_expr(st)) {
             if (((jl_expr_t*)st)->head == enter_sym) {
-                int last = (int)slength-1;
+                int last = (int)slength - 1;
                 std::set<int> as =
-                    assigned_in_try(stmts, i+1,
-                                    jl_unbox_long(jl_exprarg(st,0)), &last);
-                for(int j=0; j < (int)slength; j++) {
+                    assigned_in_try(stmts, i + 1,
+                                    jl_unbox_long(jl_exprarg(st, 0)), &last);
+                for (int j = 0; j < (int)slength; j++) {
                     if (j < i || j > last) {
                         std::set<int>::iterator it = as.begin();
-                        for(; it != as.end(); it++) {
-                            if (local_var_occurs(jl_array_ptr_ref(stmts,j), *it)) {
+                        for (; it != as.end(); it++) {
+                            if (local_var_occurs(jl_array_ptr_ref(stmts, j), *it)) {
                                 jl_varinfo_t &vi = slots[*it];
-                                if (!vi.value.constant)
-                                    vi.isVolatile = true;
+                                vi.isVolatile = true;
```

Comment:
So this change goes along with the associated small change in `emit_local()`, and it's otherwise harmless to have a constant also marked as volatile?